pkg/components/prometheus-operator/component.go

```diff
@@ -45,10 +45,36 @@ type CoreDNS struct {
 	Selector map[string]string `hcl:"selector,optional"`
 }
 
+// Ingress is a generic object for specifying Kubernetes ingress manifest.
+type Ingress struct {
```

Comment:
This should go to the separate package. Maybe just `pkg/components` for now?

Reply:
Moved to the existing common types dir.